implemented protecting routes

adamski83 opened the pull request.

adamski83 committed:
implemented protecting routes

adamski83 committed:
implementation QR code and simple routing

fengerkamil commented on src/components/form/EditFormController.tsx:
```diff
@@ -0,0 +1,109 @@
+  deleteItemHandler,
+  item,
+  toggleEdit,
+}: any) => {
```
A to dlaczego any? :(

fengerkamil commented on src/components/form/EditFormController.tsx:
```diff
@@ -0,0 +1,109 @@
+          <Controller
+            name="amount"
+            control={control}
+            defaultValue={0}
```
Masz zadeklarowane defaultValues tutaj oraz w hooku useForm, prawdopodobnie wystarczy to tylko w useFormie zadeklarować z dobrymi nazwami pól.

fengerkamil commented on src/components/form/FormController.tsx:
```diff
@@ -0,0 +1,58 @@
+  item,
+  deleteItemHandler,
+  toggleEdit,
+}: any) => {
```
dlaczego any :(

fengerkamil commented on src/components/invoices/QrScanner.tsx:
```diff
@@ -0,0 +1,26 @@
+const QrScanner: React.FC<QrScannerProps> = ({ onScan }) => {
+  const navigate = useNavigate();
+  const handleScan = (data: string | null) => {
+    if (data) {
+      onScan(data);
+      navigate(`/edit/${data}`);
+    }
+  };
```
if !data mozemy wyswietlic jakies powiadomienie lub error message ze nie udalo sie zczytać danych.

fengerkamil commented on src/components/invoices/QrScanner.tsx:
```diff
@@ -0,0 +1,26 @@
+    }
+  };
+  return (
+    <>
```
Fragment zbędny

fengerkamil commented on src/helpers/generatePdf.ts:
```diff
@@ -0,0 +1,22 @@
+export const generatePDF = (data: MockData) => {
+  const doc = new jsPDF();
+  const tableColumn = ["Name of the Box", "Description", "Amount"];
+  const tableRows: any[] = [];
```
dlaczego any :(

adamski83 committed:
Refactor form components and update dependencies

- Removed default values for form fields in AddItemForm and EditFormController components
- Updated dependencies in package.json
- Added FormControllerProps interface to types.ts
- Created useUpdateAmount and useBoxesByID hooks for updating and fetching box data
- Generated PDF now uses correct data structure for tableRows

adamski83 committed:
fixing Forms, default data and uncontroled changes
